### PR TITLE
105: Replace CLI with Python API and use PTQ nncf API

### PR DESCRIPTION
**Pull request by itrushkin**
- Replacing `mo` CLI usage with `convert_model` function call.
- Usage of HuggingFace model, dataset and evaluation.
- Replacing `POT` quantization with `nncf` API.

### CODE REVIEW

**Comment by eaidova** [comment]
@itrushkin please resolve flake8 found issues (unused imports, number of new lines before and after functions and classes definition) and add some explanation for PTQ usage steps (now you have not headers and quantization performed in section named Convert ONNX model to IR)

File: notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb [hunk 1]
```diff
@@ -18,6 +18,16 @@
     "- Compare the performance of the original, converted and quantized models."
```
Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #9.            input='input_ids[1 128],attention_mask[1 128],token_type_ids[1 128]',
</pre><p>no need in input and output specification in conevrt_model if there is no difference between onnx and it (previously input used to turn dynamic inputs to static</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Comment by itrushkin** [comment]
> @itrushkin please resolve flake8 found issues (unused imports, number of new lines before and after functions and classes definition) and add some explanation for PTQ usage steps (now you have not headers and quantization performed in section named Convert ONNX model to IR)

One `flake8` error gets me confused:
```
notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:cell_9:0:1: E303 too many blank lines (3)
```
There are no 3 blank lines in a notebook.

**Comment by itrushkin** [comment]
> > @itrushkin please resolve flake8 found issues (unused imports, number of new lines before and after functions and classes definition) and add some explanation for PTQ usage steps (now you have not headers and quantization performed in section named Convert ONNX model to IR)
> 
> One `flake8` error gets me confused:
> 
> ```
> notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:cell_9:0:1: E303 too many blank lines (3)
> ```
> 
> There are no 3 blank lines in a notebook.

The error actually indicates that a cell ends with a newline. Fixed.

File: notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb [hunk 2]
```diff
@@ -7,8 +7,8 @@
     "id": "rQc-wXjqrEuR"
```
Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #1.    def create_data_source():
</pre><p>I believe as previously, it should be in prepare dataset section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <p>there is no import dependencies part (it is above)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <p>please provide explanation. Something like:</p><p>To run this tutorial, you will need to download the General Language Understanding Evaluation  (GLUE) data for the MRPC task from HuggingFace datasets. The code below load validation split of BERTTokenizer.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <p>please add small explanation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #1.    core = ov.Core()
</pre><p>please add some comments to code for better explanation steps</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #6.    def transform_fn(data_item):
</pre><p>I think we need to add info about nncf.Dataset and transform_fn.</p><p>Also I think it is necessary explain why model_type parameter used</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #1.    def validate(model, dataset):
</pre><p>please provide type hints and doc string</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #8.            model_name=f'{ir_model_xml}.stem',
</pre><p>please remove model_name (it used in CLI for IR serialization name)</p><p>Also ir_model_bin is not used in code anymore</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #26.    print(f'F1 score: {metric}')
</pre><p>possibly it can make sense to add formating with 2 digits after coma</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (eaidova) [inline notebooks/105-language-quantize-bert/105-language-quantize-bert.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #4.    for input_layer in model.inputs:
</pre><p>Sorry, I know that it is not your code, but could you please fix?</p><p> </p><p>instead of reshaping model in cycle for each input layer here should be:</p><p> </p><p> </p><p><code>dynamic_shapes = {}</code></p><p><code>for input_layer in model.inputs:</code></p><p><code>&nbsp;&nbsp;&nbsp;input_shape = input_layer.partial_shape</code></p><p><code>&nbsp;&nbsp;&nbsp;input_shape[1] = -1</code></p><p><code>dynamic_shapes[input_layer] = input_shape</code></p><p><code>model.reshape(dynamic_shapes)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/941/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

**Review by eaidova — APPROVED** [review]
